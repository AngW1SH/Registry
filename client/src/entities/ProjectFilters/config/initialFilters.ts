import { Filters } from "../types/types";

export const initialFilters: Filters = {
  status: "Все",
  enrollmentStart: null,
  enrollmentEnd: null,
  dateStart: null,
  dateEnd: null,
  tags: null,
  text: "",
};

export const ProjectStatus = [
  "Все",
  "Активные",
  "Завершённые",
  "С вакансиями",
] as const;
