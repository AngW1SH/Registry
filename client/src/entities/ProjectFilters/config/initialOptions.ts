export type MultiselectFilterData = {
  label: string;
  options: string[];
};

export type DetailedFiltersData = {
  status: MultiselectFilterData;
  supervisor: MultiselectFilterData;
  year: MultiselectFilterData;
  format: MultiselectFilterData;
};

export const detailedFiltersInitialData: DetailedFiltersData = {
  status: {
    label: "Статус активности",
    options: ["Все", "Активные", "Завершённые", "С вакансиями"],
  },
  supervisor: {
    label: "ФИО руководителя",
    options: [
      "Иванов Павел Михайлович",
      "Гордеев Сергей Алексеевич",
      "Волков Игорь Иванович",
    ],
  },
  year: {
    label: "Курс",
    options: ["2 курс", "3 курс", "Магистратура"],
  },
  format: {
    label: "Тип занятости",
    options: ["Очный", "Удалённый", "Комбинированный"],
  },
};
