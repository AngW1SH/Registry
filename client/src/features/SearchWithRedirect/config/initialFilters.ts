import { Filters } from "../types/types";

export const initialFilters: Filters = {
  status: "Все",
  enrollDate: null,
  projectDate: null,
};

export const ProjectStatus = [
  "Все",
  "Активные",
  "Завершённые",
  "С вакансиями",
  "Доступные к покупке",
] as const;

export const ProjectStatusValues = [""];
