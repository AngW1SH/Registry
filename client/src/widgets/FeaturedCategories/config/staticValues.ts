import { CategoryCard } from "../types/types";

export const staticValues: CategoryCard[] = [
  {
    type: "image",
    name: "Математическое моделирование",
    image: "/category-bg-1.jpg",
    link: "/",
    projectsCount: 39,
  },
  {
    type: "detailed",
    name: "Математический анализ",
    showMore: true,
    link: "/",
    tags: [
      {
        id: "1",
        name: "Дифференциальные методы",
        projectsCount: 79,
      },
      {
        id: "2",
        name: "Теория чисел",
        projectsCount: 34,
      },
      {
        id: "3",
        name: "Прикладная математика",
        projectsCount: 17,
      },
      {
        id: "4",
        name: "Математическая философия",
        projectsCount: 364,
      },
      {
        id: "5",
        name: "Функциональный анализ",
        projectsCount: 34,
      },
      {
        id: "6",
        name: "Практикум",
        projectsCount: 10,
      },
    ],
  },
  {
    type: "detailed",
    name: "Машинное обучение",
    link: "/",
    showMore: false,
    tags: [
      {
        id: "7",
        name: "Робототехника",
        projectsCount: 749,
      },
      {
        id: "8",
        name: "Искусственный интеллект",
        projectsCount: 34,
      },
      {
        id: "9",
        name: "Системы управления",
        projectsCount: 14,
      },
      {
        id: "10",
        name: "Программирование",
        projectsCount: 366,
      },
      {
        id: "11",
        name: "Практикум",
        projectsCount: 10,
      },
    ],
  },
  {
    type: "image",
    name: "Web разработка",
    image: "/category-bg-4.jpg",
    link: "/",
    projectsCount: 125,
  },
  {
    type: "detailed",
    name: "История",
    link: "/",
    showMore: false,
    tags: [
      {
        id: "12",
        name: "История XV-XVI вв",
        projectsCount: 79,
      },
      {
        id: "13",
        name: "История XX в",
        projectsCount: 34,
      },
      {
        id: "14",
        name: "История V-XVI вв",
        projectsCount: 17,
      },
      {
        id: "15",
        name: "Древний восток",
        projectsCount: 79,
      },
      {
        id: "16",
        name: "История XIX в",
        projectsCount: 34,
      },
      {
        id: "17",
        name: "История V-XIV в",
        projectsCount: 17,
      },
    ],
  },
  {
    type: "detailed",
    name: "Биология",
    link: "/",
    showMore: true,
    tags: [
      {
        id: "27",
        name: "Генетика",
        projectsCount: 79,
      },
      {
        id: "19",
        name: "Клеточная биология",
        projectsCount: 34,
      },
      {
        id: "20",
        name: "Протеомика",
        projectsCount: 17,
      },
      {
        id: "21",
        name: "Фармакология",
        projectsCount: 364,
      },
      {
        id: "31",
        name: "Секвенирование",
        projectsCount: 34,
      },
    ],
  },
  {
    type: "image",
    name: "Юриспруденция",
    link: "/",
    image: "/category-bg-7.jpg",
    projectsCount: 39,
  },
  {
    type: "detailed",
    name: "Медицинская генетика",
    showMore: true,
    link: "/",
    tags: [
      {
        id: "18",
        name: "ДНК, РНК",
        projectsCount: 79,
      },
      {
        id: "19",
        name: "Клеточная биология",
        projectsCount: 34,
      },
      {
        id: "20",
        name: "Протеомика",
        projectsCount: 17,
      },
      {
        id: "21",
        name: "Фармакалогия",
        projectsCount: 364,
      },
      {
        id: "22",
        name: "Секвенирование",
        projectsCount: 34,
      },
      {
        id: "23",
        name: "Клеточная биология",
        projectsCount: 10,
      },
    ],
  },
  {
    type: "image",
    name: "Финансы и право",
    link: "/",
    image: "/category-bg-9.jpg",
    projectsCount: 39,
  },
  {
    type: "detailed",
    name: "Социология",
    link: "/",
    showMore: false,
    tags: [
      {
        id: "24",
        name: "Общество",
        projectsCount: 79,
      },
      {
        id: "25",
        name: "Языки",
        projectsCount: 34,
      },
      {
        id: "26",
        name: "Социальные тенденции",
        projectsCount: 17,
      },
    ],
  },
];
