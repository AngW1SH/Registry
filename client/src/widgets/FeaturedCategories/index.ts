import FeaturedCategories from "./ui/FeaturedCategories";

export { FeaturedCategories };
