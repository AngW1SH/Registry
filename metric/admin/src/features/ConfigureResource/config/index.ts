import { PlatformName } from "@/entities/Platform/types";
import { ResourceConfig } from "../types";
import { githubConfig } from "./github";

export const configs: { [key in PlatformName]: ResourceConfig } = {
  GitHub: githubConfig,
};
