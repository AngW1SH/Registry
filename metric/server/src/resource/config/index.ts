import { PlatformName } from 'src/platform/platform.entity';
import { ResourceConfig } from './types';
import { githubConfig } from './platforms/github';

export const configs: { [key in PlatformName]: ResourceConfig } = {
  GitHub: githubConfig,
  GitLab: githubConfig,
};
