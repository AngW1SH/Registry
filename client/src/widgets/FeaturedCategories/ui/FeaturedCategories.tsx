import { FC } from "react";
import SuggestProject from "./SuggestProject";
import CategoryCardDetailed from "./CategoryCardDetailed";
import CategoryCardWithImage from "./CategoryCardWithImage";
import { getCategoriesData } from "../api/getCategoriesData";
import { CategoryCard } from "../types/types";
import { staticValues } from "../config/staticValues";

interface FeaturedCategoriesProps {
  categories?: CategoryCard[];
}

const FeaturedCategories: FC<FeaturedCategoriesProps> = async ({
  categories,
}) => {
  const values = categories ? categories : await getCategoriesData();

  return (
    <div className="grid grid-cols-1 gap-2 sm:grid-cols-2 sm:grid-rows-6 lg:grid-cols-3 lg:grid-rows-4 xl:grid-cols-4 xl:grid-rows-3">
      {values.map((category) =>
        category.type == "detailed" ? (
          <CategoryCardDetailed
            key={category.name}
            className="overflow-hidden rounded-lg"
            category={category}
          />
        ) : (
          <CategoryCardWithImage
            key={category.name}
            className="overflow-hidden rounded-lg"
            category={category}
          />
        ),
      )}
      <SuggestProject className="col-span-1 row-span-2 overflow-hidden rounded-lg sm:col-start-2 sm:row-start-2 lg:col-start-3 xl:col-start-4" />
    </div>
  );
};

export default FeaturedCategories;
