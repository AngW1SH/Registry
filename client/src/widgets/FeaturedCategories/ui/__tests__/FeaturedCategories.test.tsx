import { render, act } from "@testing-library/react";
import "@testing-library/jest-dom";
import FeaturedCategories from "../FeaturedCategories";
import { staticValues } from "../../config/staticValues";

jest.mock("next/navigation");

describe("FeaturedCategories Widget UI", () => {
  it('renders a "suggest project" banner', async () => {
    const Component = FeaturedCategories({ categories: staticValues });
    // @ts-ignore
    const { getByText } = render(await Component);

    expect(getByText(/Предложить тему проекта/)).toBeInTheDocument();
  });

  it("renders categories from the list", async () => {
    const Component = FeaturedCategories({ categories: staticValues });
    // @ts-ignore
    const { getByText } = render(await Component);

    expect(getByText(/Математический анализ/)).toBeInTheDocument();
    expect(getByText(/Математическое моделирование/)).toBeInTheDocument();
  });

  test("snapshot test", async () => {
    const Component = FeaturedCategories({ categories: staticValues });
    // @ts-ignore
    const { asFragment } = render(await Component);

    expect(asFragment).toMatchSnapshot();
  });
});
