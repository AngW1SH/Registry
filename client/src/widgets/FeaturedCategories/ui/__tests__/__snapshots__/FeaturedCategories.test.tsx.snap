// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedCategories Widget UI snapshot test 1`] = `[Function]`;
